hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/VolumeInfo.java

```diff
@@ -134,25 +135,45 @@ public VolumeInfo build() throws IOException {
   }
 
   private long getReserved(ConfigurationSource conf) {
+    final float defaultValue = Float.MAX_VALUE;
+    float percentage = conf.getFloat(HDDS_DATANODE_VOLUME_RESERVED,
+        defaultValue);
     Collection<String> reserveList = conf.getTrimmedStringCollection(
         HDDS_DATANODE_DIR_DU_RESERVED);
-    for (String reserve : reserveList) {
-      String[] words = reserve.split(":");
-      if (words.length < 2) {
-        LOG.error("Reserved space should config in pair, but current is {}",
-            reserve);
-        continue;
-      }
+    //Both the configs are set. Log it and return 0
+    if (reserveList.size() > 0 && percentage != defaultValue) {
+      LOG.error("Both {} and {} are set. Set either one, not both. " +
+          "Setting reserved to 0.", HDDS_DATANODE_VOLUME_RESERVED,
+          HDDS_DATANODE_DIR_DU_RESERVED);
+      return 0;
```

Comment:
We should fall back to the more explicit setting instead of failing the service. If the admin actually set this per volume that should be honored first IMO.

Reply:
Thanks for the suggestion. I have updated the PR.